libs/angular/src/lib/modal/modal.service.ts

```diff
@@ -0,0 +1,44 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, OnDestroy } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class NggModalService implements OnDestroy {
+    private locks: HTMLElement[] = [];
+    
+    constructor (@Inject(DOCUMENT) private document: Document) {}
+
+    public lockBody(htmlElement: HTMLElement) {
+        // Check if lock already exists
+        if (this.locks.some((l) => l === htmlElement))
+            return;
+
+        // If no locks exists, then add class
+        if (this.locks.length <= 0) {
+            this.document.body.classList.add("ngg-overflow-hidden");
```

Comment:
I cannot find any implementation of this class? Anyhows, since locking the body in different browsers/devices is a bit different I think we should go with a library to handle this. For the dropdown and datepicker we have used 'body-scroll-lock'. It should be an easy replacement :)

Reply:
Yes that class is missing, I forgot about that. In our project we placed this class in the styles.scss (toplevel/root scss-file), since we couldn't place it in the scss-file for the component because of component encapsulation.

Anyway, it sounds like a good idea to use the same library in this case as well. Will change that.